go.mod

```diff
@@ -1,142 +1,149 @@
 module github.com/siderolabs/cluster-api-control-plane-provider-talos
 
-go 1.19
+go 1.20
 
 require (
-	github.com/coreos/go-semver v0.3.0
+	github.com/coreos/go-semver v0.3.1
 	github.com/go-logr/logr v1.2.3
 	github.com/go-logr/zapr v1.2.3 // indirect
+	github.com/gobuffalo/flect v1.0.2
 	github.com/google/uuid v1.3.0
-	github.com/onsi/ginkgo v1.16.5
-	github.com/onsi/gomega v1.24.1
+	github.com/onsi/gomega v1.27.5
 	github.com/pkg/errors v0.9.1
 	github.com/siderolabs/capi-utils v0.0.0-20221215150410-835519e95d9c
 	github.com/siderolabs/cluster-api-bootstrap-provider-talos v0.5.6
+	github.com/siderolabs/crypto v0.4.0
 	github.com/siderolabs/go-retry v0.3.2
-	github.com/siderolabs/talos/pkg/machinery v1.3.0
-	github.com/stretchr/testify v1.8.1
-	google.golang.org/grpc v1.51.0
+	github.com/siderolabs/talos v1.3.5
+	github.com/siderolabs/talos/pkg/machinery v1.4.0
+	github.com/stretchr/testify v1.8.2
+	google.golang.org/grpc v1.54.0
+	gopkg.in/typ.v4 v4.2.0
 	gopkg.in/yaml.v3 v3.0.1
-	k8s.io/api v0.25.0
-	k8s.io/apimachinery v0.25.0
-	k8s.io/apiserver v0.25.0
-	k8s.io/client-go v0.25.0
-	k8s.io/utils v0.0.0-20220823124924-e9cbc92d1a73
-	sigs.k8s.io/cluster-api v1.3.1
-	sigs.k8s.io/controller-runtime v0.13.1
+	k8s.io/api v0.26.1
+	k8s.io/apiextensions-apiserver v0.26.1
+	k8s.io/apimachinery v0.26.1
+	k8s.io/apiserver v0.26.1
+	k8s.io/client-go v0.26.1
+	k8s.io/klog/v2 v2.90.1
+	k8s.io/utils v0.0.0-20230209194617-a36077c30491
+	sigs.k8s.io/cluster-api v1.4.1
```

Comment:
we should update controller-gen/conversion-gen versions as well  - cluster-api repo has version, or you can take a look at CABPT